internal/binary/envoy/install_test.go

```diff
@@ -90,16 +85,47 @@ func TestUntarEnvoy(t *testing.T) {
 	require.Contains(t, out.String(), `100% |████████████████████████████████████████|`)
 }
 
-func TestFetchIfNeeded_ErrorOnIncorrectURL(t *testing.T) {
+func TestInstallIfNeeded_ErrorOnIncorrectURL(t *testing.T) {
 	o, cleanup := setupTest(t)
 	defer cleanup()
 
 	o.ManifestURL += "/mannyfest.json"
-	_, e := FetchIfNeeded(&o.GlobalOpts, o.reference)
+
+	_, e := InstallIfNeeded(&o.GlobalOpts, runtime.GOOS, version.Envoy)
 	require.EqualError(t, e, "received 404 status code from "+o.ManifestURL)
 	require.Empty(t, o.Out.(*bytes.Buffer))
 }
 
+func TestInstallIfNeeded_Validates(t *testing.T) {
```

Comment:
this test took me a few minutes figure out what these cases actually test.. the difficult part is that the manifest setup here in `setupTest` is not defined in this _test.go but declared in the nested `RequireManifestTestServer`.. Can we improve here? e.g. having `RequireManifestTestServer(t *testing.T, m *manifest.Manifest)` signature for `RequireManifestTestServer`?

Reply:
I really hate spending even more time on manifest server as it will be deleted in the next PR. can we stop maybe?

Reply:
maybe this wasn't in the same file before, but setupTest is in the same file now for sure.. I renamed it setupInstallTest to make it more clear and did a pass to ensure nothing else is importing it.